catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java

```diff
@@ -13,139 +13,756 @@
+                if (query.view.equals(NsiliConstants.NSIL_ASSOCIATION_VIEW)) {
+                    shouldRun = false;
+                }
+
+                if (shouldRun && !paused) {
+                    if (standingQueryData.size() <= maxPendingResults) {
+                        DAGQueryResult queryResult = getData(queryTime);
+                        if (queryResult != null) {
+                            standingQueryData.add(queryResult);
+                        }
+                    }
+
+                    if (standingQueryData.size() > 0) {
+                        List<Callback> failedCallbacks = new ArrayList<>();
+                        for (Callback callback : callbacks.values()) {
+                            try {
+                                if (standingQueryData.size() > 0) {
+                                    LOGGER.trace("Notifying callback that results are available: {}",
+                                            standingQueryData.size());
+                                    callback._notify(org.codice.alliance.nsili.common.UCO.State.RESULTS_AVAILABLE,
+                                            get_request_description());
+                                }
+                            } catch (InvalidInputParameter invalidInputParameter) {
+                                LOGGER.warn("Unable to notify callback", invalidInputParameter);
+                            } catch (ProcessingFault processingFault) {
+                                LOGGER.warn("Unable to notify callback", processingFault);
+                            } catch (SystemFault systemFault) {
+                                LOGGER.warn("Unable to notify callback", systemFault);
+                            } catch (Exception e) {
+                                failedCallbacks.add(callback);
+                            }
+                        }
+
+                        failedCallbacks.stream()
+                                .forEach(c -> freeCallback(c));
+                    }
+                    lastCompletedExecutionTime = System.currentTimeMillis();
+                }
+
+                //Don't sleep if more results available. Need client to pick up results as fast
+                //as possible to catch up.
+                if (!executionThread.hasMoreResultsAvailOnLastQuery()) {
+                    try {
+                        Thread.sleep(updateRate);
+                    } catch (Exception ignore) {
+                    }
+                }
+            }
+
+            synchronized (callbackLockObj) {
+                callbacks.clear();
+            }
+        }
+
+        protected DAGQueryResult getData(long queryTime) {
+            DAGQueryResult result = null;
+
+            List<Result> catalogResults = new ArrayList<>();
+
+            Filter parsedFilter = bqsFilter;
+            if (!moreResultsAvailOnLastQuery && queryTime > 0) {
+                parsedFilter = filterBuilder.allOf(bqsFilter,
```

Comment:
Right now they are not handled. For local data we could probably look at the original standing query request and modify it to look at history and find cards where the last entry is deleted. For external sources, I'm not sure what we would do there. Each time through the StandingQuery interface modifies the time criteria to restrict the query to only results that would not have been already returned.